app/src/main/java/com/duckduckgo/app/browser/downloader/NetworkFileDownloader.kt

```diff
@@ -32,13 +34,25 @@ import javax.inject.Inject
 class NetworkFileDownloader @Inject constructor(
     private val context: Context,
     private val filenameExtractor: FilenameExtractor,
-    private val fileService: DownloadFileService
+    private val fileService: DownloadFileService,
+    private val appBuildConfig: AppBuildConfig
 ) {
 
     fun download(
         pendingDownload: PendingFileDownload,
         callback: FileDownloader.FileDownloadListener
     ) {
+        if (appBuildConfig.flavor == INTERNAL) {
+            downloadV2(pendingDownload, callback)
+        } else {
+            downloadV1(pendingDownload, callback)
+        }
+    }
+
+    fun downloadV1(
```

Comment:
This is the original `download` version. Renamed to `downloadV1` as `downloadV2` was added and kept the public `download` signature unchanged.